eitprocessing/binreader/reader.py

```diff
@@ -43,29 +45,44 @@ def read_string(self, length=1):
         data = data[0].decode().rstrip()
         return data
 
-    def _read_full_type_code(self, full_type_code) -> tuple[Any, ...]:
+    def _read_full_type_code(self, full_type_code, endian: str = None) -> tuple[Any, ...]:
+        if endian:
+            if endian in ['little', 'big']:
+                full_type_code = '<' + full_type_code if endian == 'little' else '>' + full_type_code
+            else:
+                warnings.warn('Endian type not recognized. Allowed values are '
+                              '\'little\' and \'big\'')
         data_size = struct.calcsize(full_type_code)
         packed_data = self.file_handle.read(data_size)
         data = struct.unpack(full_type_code, packed_data)
         return data
 
-    def float32(self) -> float:
-        return self.read_single(type_code="f", cast=float)
+    def float32(self, endian: str = None) -> float:
+        return self.read_single(type_code="f", cast=float, endian=endian)
 
-    def float64(self) -> float:
-        return self.read_single(type_code="d", cast=float)
+    def float64(self, endian: str = None) -> float:
+        return self.read_single(type_code="d", cast=float, endian=endian)
 
-    def npfloat32(self, length=1) -> NDArray[np.float32]:
-        return self.read_array(type_code="f", cast=np.float32, length=length)
+    def npfloat32(self, length=1, endian: str = None) -> NDArray[np.float32]:
+        return self.read_array(type_code="f", cast=np.float32, length=length, endian=endian)
 
-    def npfloat64(self, length=1) -> NDArray[np.float64]:
-        return self.read_array(type_code="d", cast=np.float64, length=length)
+    def npfloat64(self, length=1, endian: str = None) -> NDArray[np.float64]:
+        return self.read_array(type_code="d", cast=np.float64, length=length, endian=endian)
 
-    def int32(self) -> int:
-        return self.read_single(type_code="i", cast=int)
+    def int32(self, endian: str = None) -> int:
+        return self.read_single(type_code="i", cast=int, endian=endian)
 
-    def npint32(self, length=1) -> NDArray[np.int32]:
-        return self.read_array(type_code="i", cast=np.int32, length=length)
+    def npint32(self, length=1, endian: str = None) -> NDArray[np.int32]:
+        return self.read_array(type_code="i", cast=np.int32, length=length, endian=endian)
 
     def string(self, length=1) -> str:
         return self.read_string(length=length)
+
+    def unsigned_char(self, endian: str = None) -> int:
```

Comment:
These names are incosistent with the previously used names.

```python
    def int8(self, endian: str = None) -> int:
```

Reply:
The struct documentation lists the C type. The Python type for these methods is either int, float, or str. It is not clear to me from `unsigned_char()` that this becomes an int type. `uint8` conforms to the naming convention numpy uses (https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.int8). I would argue that we should use the Python type name and numpy naming conventions.

Reply:
I agree, using the numpy naming convention sounds very logical. I will change the naming.